File: 0844-backspace-string-compare/0844-backspace-string-compare.cpp
class Solution {
public:
    bool backspaceCompare(string s, string t) {
        string newS = "", newT = "";
        int count = 0;
        for(int i = s.length()-1; i >= 0; i--){
            if(s[i] == '#') count++;
            else{
                if(count == 0) newS = s[i]+newS;
                else{
                    count--;
                }
            }
        }
        count = 0;
        for(int i = t.length()-1; i >= 0; i--){
            if(t[i] == '#') count++;
            else{
                if(count == 0) newT = t[i]+newT;
                else{
                    count--;
                }
            }
        }
        return newS==newT;
    }
};

File: 0844-backspace-string-compare/0844-backspace-string-compare.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        s_new, t_new = "", ""
        count = 0
        for i in range (len(s)-1, -1, -1):
            if s[i] == '#':
                count += 1
            else:
                if count == 0:
                    s_new = s[i] + s_new
                else:
                    count -= 1
        count = 0
        for i in range (len(t)-1, -1, -1):
            if t[i] == '#':
                count += 1
            else:
                if count == 0:
                    t_new = t[i] + t_new
                else:
                    count -= 1
        return s_new == t_new


        